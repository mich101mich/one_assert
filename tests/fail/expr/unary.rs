fn main() {
    one_assert::assert!(*false);
    one_assert::assert!(-false);
    one_assert::assert!(*1);
    one_assert::assert!(!0);
    one_assert::assert!(-0);
}
