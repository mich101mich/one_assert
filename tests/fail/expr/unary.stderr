error[E0614]: type `bool` cannot be dereferenced
 --> tests/fail/expr/unary.rs:2:25
  |
2 |     one_assert::assert!(*false);
  |                         ^^^^^^

error[E0600]: cannot apply unary operator `-` to type `bool`
 --> tests/fail/expr/unary.rs:3:25
  |
3 |     one_assert::assert!(-false);
  |                         ^^^^^^ cannot apply unary operator `-`

error[E0614]: type `{integer}` cannot be dereferenced
 --> tests/fail/expr/unary.rs:4:25
  |
4 |     one_assert::assert!(*1);
  |                         ^^

error[E0308]: mismatched types
 --> tests/fail/expr/unary.rs:5:25
  |
5 |     one_assert::assert!(!0);
  |                         ^^ expected `bool`, found integer

error[E0308]: mismatched types
 --> tests/fail/expr/unary.rs:6:25
  |
6 |     one_assert::assert!(-0);
  |                         ^^ expected `bool`, found integer
